doc/modules/neighbors.rst

```diff
@@ -514,3 +514,224 @@ the model from 0.81 to 0.82.
+   :scale: 50
+
+.. |nca_classification_2| image:: ../auto_examples/neighbors/images/sphx_glr_plot_nca_classification_002.png
+   :target: ../auto_examples/neighbors/plot_nca_classification.html
+   :scale: 50
+
+.. centered:: |nca_classification_1| |nca_classification_2|
+
+The plot shows decision boundaries for Nearest Neighbor Classification and
+Neighborhood Components Analysis classification, when training and scoring
+on only two features, for visualisation purpose.
+
+
+Dimensionality reduction
+------------------------
+
+:class:`NeighborhoodComponentsAnalysis` can be used to perform supervised
+dimensionality reduction. The input data are projected onto a linear subspace
+consisting of the directions which minimize the NCA objective. The desired
+dimensionality can be set using the parameter ``n_features_out``. For instance,
+the following shows a comparison of dimensionality reduction with Principal
+Component Analysis (:class:`sklearn.decomposition.PCA`), Linear Discriminant
+Analysis (:class:`sklearn.discriminant_analysis.LinearDiscriminantAnalysis`)
+and Neighborhood Component Analysis (:class:`NeighborhoodComponentsAnalysis`)
+on the Digits dataset, a dataset with size :math:`n_{samples} = 1797` and
+:math:`n_{features} = 64`. The data set is splitted in a training and test set
+of equal size. What is more, a :class:`sklearn.preprocessing.StandardScaler`
+fitted on the training set and transforms the data from both sets. For
+evaluation the 3-nearest neighbor classification accuracy is computed on the
+2-dimensional embedding found by each method. Each data sample belongs to one
+of 10 classes.
+
+.. |nca_dim_reduction_1| image:: ../auto_examples/neighbors/images/sphx_glr_plot_nca_dim_reduction_001.png
+   :target: ../auto_examples/neighbors/plot_nca_dim_reduction.html
+   :width: 32%
+
+.. |nca_dim_reduction_2| image:: ../auto_examples/neighbors/images/sphx_glr_plot_nca_dim_reduction_002.png
+   :target: ../auto_examples/neighbors/plot_nca_dim_reduction.html
+   :width: 32%
+
+.. |nca_dim_reduction_3| image:: ../auto_examples/neighbors/images/sphx_glr_plot_nca_dim_reduction_003.png
+   :target: ../auto_examples/neighbors/plot_nca_dim_reduction.html
+   :width: 32%
+
+.. centered:: |nca_dim_reduction_1| |nca_dim_reduction_2| |nca_dim_reduction_3|
+
+
+Mathematical formulation
+------------------------
+
+NCA learns a linear transformation matrix :math:`L` of
+size ``(n_features_out, n_features)``, which maximises in average the
+probability :math:`p_i` of sample :math:`i` being
+classified as :math:`C_i`, defined by:
+
+.. math::
+
+  p_{i}=\sum\nolimits_{j \in C_i}{p_{i j}}
+
+where :math:`C_i` is the set of points in the same class as sample :math:`i`,
+and :math:`p_{i j}` is the softmax over Euclidean distances in the
+transformed space:
+
+.. math::
+
+  p_{i j} = \frac{\exp(-||L x_i - L x_j||^2)}{\sum\nolimits_{k \ne
+            i} {\exp{-(||L x_i - L x_k||^2)}}} , p_{i i} = 0
```

Comment:
could add \quad before p_{i i} = 0 to improve readability

Reply:
Done.